test/unit/async/future.cpp

```diff
@@ -489,6 +489,9 @@ TYPED_TEST(Error, Simple2) {
   tp->Wait();
 }
 
+// race condition on CurrentThreadPool for fibers
```

Comment:
You should implement
YACLIB_THREAD_LOCAL(name) = 
in common threads it is thread_local
in fibers its proxy object, that contain iterator to global hashmap:
hashmap<thread_it, hashmap<std::string_view, value>>